src/types/cql-patient.interfaces.ts

```diff
@@ -1,11 +1,26 @@
-import { Record } from '../cql-patient';
-
+/*
+ * Iterator for the patients provided to the execution engine
+ */
 export interface DataProvider {
   patients: any[];
```

Comment:
I don't believe this 'patients' array is expected by the engine at all. This may be a pretty hefty request of some DataProvider implementations as well.

Reply:
Good point, @hossenlopp.  I totally missed that one (even though it's very similar to my comment about `TerminologyProvider`'s `valueSets`).  You, @mgramigna, and I make a good team.  We'll eventually get this (mostly) right!

Reply:
good call!